### PR TITLE
Simulation: Execution trace (PC/Stack/Scratch) support

### PR DESCRIPTION
**Pull request by ahangsu**

### CODE REVIEW

File: .test-env [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # Configs for testing repo download:
 SDK_TESTING_URL="https://github.com/algorand/algorand-sdk-testing"
-SDK_TESTING_BRANCH="master"
+SDK_TESTING_BRANCH="simulate-exec-trace-stack-scratch"
```
Comment (ahangsu) [inline .test-env:3]:
```suggestion
SDK_TESTING_BRANCH="master"
```

File: algosdk/v2client/models/simulate_request.py [hunk 2]
```diff
@@ -16,11 +16,36 @@ def dictify(self) -> Dict[str, Any]:
         return {"txns": [txn.dictify() for txn in self.txns]}
 
 
+class SimulateTraceConfig(object):
```
Comment (tzaffi) [inline algosdk/v2client/models/simulate_request.py:19]:
```suggestion
class SimulateTraceConfig:
```

**Review by tzaffi — COMMENTED** [review]
I know it's still a draft but couldn't help noticing an itsy bitsy nit.

File: algosdk/v2client/models/simulate_request.py (hunk 2 above)
Comment (ahangsu) [reply algosdk/v2client/models/simulate_request.py:19]: right, I can move that later, just mucking around and warming up with SDKs.

File: algosdk/v2client/models/simulate_request.py [hunk 3]
```diff
@@ -29,11 +62,13 @@ def __init__(
         allow_more_logs: bool = False,
         allow_empty_signatures: bool = False,
         extra_opcode_budget: int = 0,
+        exec_trace_config: SimulateTraceConfig = SimulateTraceConfig(),
```
Comment (jasonpaulos) [inline algosdk/v2client/models/simulate_request.py:65]: minor: I'd prefer if this defaults to None, then in the function body you call the `SimulateTraceConfig` constructor

File: tests/steps/other_v2_steps.py [hunk 4]
```diff
@@ -1530,6 +1531,111 @@ def power_pack_simulation_should_have_extra_budget(context, budget):
...
+        enable=True,
+        stack_change="stack" in option_list,
+        scratch_change="scratch" in option_list,
+    )
+
+
+@then(
+    '{unit_index}th unit in the "{trace_type}" trace at txn-groups path "{group_path}" should add value "{stack_addition:MaybeString}" to stack, pop {pop_count} values from stack, write value "{scratch_var:MaybeString}" to scratch slot "{scratch_index:MaybeString}".'
+)
+def step_impl(
```
Comment (jasonpaulos) [inline tests/steps/other_v2_steps.py:1549]: nit: rename function?

File: tests/steps/other_v2_steps.py [hunk 5]
```diff
@@ -1530,6 +1531,111 @@ def power_pack_simulation_should_have_extra_budget(context, budget):
...
+        if avm_type == "uint64":
+            assert avm_value["type"] == 2
+            if int(value) > 0:
+                assert avm_value["uint"] == int(value)
+            else:
+                assert "uint" not in avm_value
+        elif avm_value == "bytes":
+            assert avm_value["type"] == 1
+            if len(value) > 0:
+                assert avm_value["bytes"] == base64.b32decode(bytearray(value))
```
Comment (jasonpaulos) [inline tests/steps/other_v2_steps.py:1605]: shouldn't this be `b64decode`?

**Review by jasonpaulos — COMMENTED** [review]
For the most part looks good I just have some minor comments

File: algosdk/v2client/models/simulate_request.py (hunk 3 above)
Comment (ahangsu) [reply algosdk/v2client/models/simulate_request.py]: see [pr comment response](https://github.com/algorand/py-algorand-sdk/pull/505/commits/9c9baade970c1cdae430a231df7cdd8279464fb9)

File: tests/steps/other_v2_steps.py (hunk 5 above)
Comment (ahangsu) [reply tests/steps/other_v2_steps.py]: good catch, my typo, fixed in [pr comment response](https://github.com/algorand/py-algorand-sdk/pull/505/commits/9c9baade970c1cdae430a231df7cdd8279464fb9)

**Review by jasonpaulos — APPROVED** [review]

**Review by bbroder-algo — APPROVED** [review]

**Pull request merged**